File: Core/src/main/java/dev/lynith/core/ui/styles/impl/CornerRadius.java
package dev.lynith.core.ui.styles.impl;

import dev.lynith.core.ui.styles.Style;

public class CornerRadius extends Style<CornerRadius, Integer> {

    private Integer value;

    public CornerRadius() {
        this(0);
    }

    public CornerRadius(Integer value) {
        this.value = value;
    }

    @Override
    public Integer getValue() {
        return value;
    }

    @Override
    public void setValue(Integer value) {
        this.value = value;
    }

    @Override
    public String getName() {
        return "corner-radius";
    }

    @Override
    public String getValueSerialized() {
        return value.toString();
    }

}

File: Core/src/main/kotlin/dev/lynith/core/ui/styles/impl/CornerRadius.kt
package dev.lynith.core.ui.styles.impl

import dev.lynith.core.ui.styles.Style

class CornerRadius @JvmOverloads constructor(
    var value: Int = 0
) : Style<CornerRadius>() {

    override val propertyName = "corner-radius"
    override val valueSerialized = value.toString()
}
